### PR TITLE
Issue 112 add phone field

### PR DESCRIPTION
**Pull request by ledenev-viktor**
Added libs for intl-phone and validations phone numbers
Added translations
Added a phone field
Fixed form validation
Fixed global and local styles

### CODE REVIEW

File: src/components/widgets/registration/steps/ContactsStep.tsx [hunk 1]
```diff
@@ -42,6 +47,27 @@ export const ContactsStep = ({
             }}
             onSaveEdits={onSaveEdits}
         >
+            <InputPhone
+                name="phone"
+                listCountries={['ge', 'us', 'ru']}
+                label={t('content_form_phone')}
+                rules={{
```
Comment (YuliaKletsova) [inline src/components/widgets/registration/steps/ContactsStep.tsx:54]: WDYT about moving rules to component. They are universal, so it is better to have it inside input-phone

**Comment by YuliaKletsova** [comment]
I think in general we should not use the library now. We have just two options and we can handle mask ourselft. Select for country code is not convinient enough

**Comment by ledenev-viktor** [comment]
> Посмотрите здесь https://github.com/faraazHasan/react-phone-input-pro#readme

очень сомнительная либа. попытался поставить, много ошибок компиляции. 
Предлагаю библиотеки с низким рейтингом не рассматривать. Т.к. они как правило имеют много недоработок

File: src/components/widgets/registration/steps/ContactsStep.tsx (hunk 1 above)
Comment (ledenev-viktor) [reply src/components/widgets/registration/steps/ContactsStep.tsx]: исправлено

File: src/components/widgets/registration/steps/ContactsStep.tsx [hunk 2]
```diff
@@ -42,6 +47,27 @@ export const ContactsStep = ({
             }}
             onSaveEdits={onSaveEdits}
         >
+            <InputPhone
+                name="phone"
+                listCountries={['ge', 'us', 'ru']}
```
[the remark this answers is not in the document]
Comment (ledenev-viktor) [reply src/components/widgets/registration/steps/ContactsStep.tsx]: исправлено

File: src/components/ui/hook-form/input-phone/index.tsx [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        return ['ge', 'us', 'ru'].slice().includes(iso2);
+    });
+
+    const [mask, setMask] = useState<string | undefined>();
+    const [iso, setIso] = useState<string | undefined>();
+
+    useEffect(() => {
+        switch (iso) {
+            case 'ge':
+                setMask('.. ... ....');
```
Comment (YuliaKletsova) [inline src/components/ui/hook-form/input-phone/index.tsx:56]: Let's use this mask '** *** ** **', also we need same mask for admin side.

File: src/styles/global-styles.ts [hunk 4]
```diff
@@ -10,7 +10,8 @@ export const globalStyles = css`
         margin: 0;
         padding: 0;
         background-color: var(--background);
-        overflow: hidden;
+        overflow-x: hidden;
+        overflow-y: visible;
```
Comment (YuliaKletsova) [inline src/styles/global-styles.ts:14]: Why did you change it? It horizontal scroll right?

File: src/components/ui/hook-form/input-phone/index.tsx [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+                        props.onBlur?.(e);
+                    }}
+                />
+                {error && <ErrorMessage motionId={name}>{error}</ErrorMessage>}
+            </AnimatePresence>
+        </Flex>
+    );
+};
+
+export const InputPhone = styled(InputPhoneBase)`
```
Comment (YuliaKletsova) [inline src/components/ui/hook-form/input-phone/index.tsx:89]: i don't think we should mix emotion with classnames, so can we choose one variant? Better if we can use classnames

File: src/components/ui/hook-form/input-phone/index.tsx [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+    const { control, trigger } = useFormContext();
+    const { field, fieldState } = useController({
+        name,
+        defaultValue,
+        control,
+        rules,
+    });
+    const error = fieldState.error ? fieldState.error.message : '';
+
+    const countries = defaultCountries.filter((country) => {
```
Comment (YuliaKletsova) [inline src/components/ui/hook-form/input-phone/index.tsx:45]: Not you calculate it for every rerender right? It's pretty much like the constant, so we can calculate it globally in this file. Just once

File: src/components/ui/hook-form/timeslots/index.tsx [hunk 7]
```diff
@@ -41,6 +42,14 @@ const TimeSlotsBase: FC<TimeSlotsBaseProps> = ({
 
     const error = errors[name] ? <>{errors[name]?.message}</> : '';
 
+    const valueField = useWatch({ control, name });
+
+    useEffect(() => {
+        if (Array.isArray(valueField) && valueField.length > 0) {
+            trigger(name);
+        }
+    }, [name, trigger, valueField]);
```
Comment (YuliaKletsova) [inline src/components/ui/hook-form/timeslots/index.tsx:51]: Just curious: why did you change it? In two words, what is the purpose of this changes

File: src/components/widgets/registration/FormComponent.tsx [hunk 8]
```diff
@@ -25,9 +26,27 @@ type RegFormBaseProps = {
...
+        const currentErrors = Object.keys(errors) as Array<keyof FormFields>;
+
+        if (currentErrors.length > 0) {
+            const timeoutId = setTimeout(() => {
+                trigger(currentErrors);
+            }, 0);
+
+            return () => clearTimeout(timeoutId);
+        }
+    }, [locale, errors, trigger]);
```
Comment (YuliaKletsova) [inline src/components/widgets/registration/FormComponent.tsx:49]: ????

File: src/components/widgets/registration/index.tsx [hunk 9]
```diff
@@ -22,7 +23,7 @@ export const RegistrationForm: FC<RegistrationFormProps> = () => {
 
     const form = useForm<FormFields>({
         defaultValues,
-        mode: 'onChange',
+        mode: 'onSubmit',
```
Comment (YuliaKletsova) [inline src/components/widgets/registration/index.tsx:26]: onBlur will be better, why onSubmit? This solution force you to trigger errors manually, i do not understand why can't we use built-in functionality in this case. Only specific fields like slots and calendar needs manual trigger in case we use onBlur

**Review by YuliaKletsova — APPROVED** [review]

**Review by YuliaKletsova — CHANGES_REQUESTED** [review]
missckicked
